romancal/stpipe/core.py

```diff
@@ -58,6 +59,25 @@ def _datamodels_open(cls, init, **kwargs):
             return ModelLibrary(init)
         raise TypeError(f"Invalid input: {init}")
 
+    @classmethod
+    def _get_crds_parameters(cls, dataset):
+        crds_parameters, crds_observatory = super()._get_crds_parameters(dataset)
+        if "roman.meta.instrument.name" not in crds_parameters:
```

Comment:
I opened an issue to track the removal of this work-around once the L3 metadata is finalized: https://github.com/spacetelescope/romancal/issues/1817